src/app/services/movie.service.ts

```diff
@@ -0,0 +1,61 @@
+import type { MovieWithFavorite } from '@interfaces/movie.interface.ts';
+import type { PaginationOptions, PaginationResponse } from '@interfaces/pagination.interface.ts';
+import { wait } from '@utils/wait.ts';
+
+import { localStorageService, type LocalStorageState, type StorageService } from './local-storage.service.ts';
+
+export class MovieService {
+  constructor(private readonly localStorageService: StorageService<LocalStorageState>) {}
+
+  public async getMovies(
+    { page, limit }: PaginationOptions,
+    isFavoriteOnly: boolean,
+  ): Promise<PaginationResponse<MovieWithFavorite>> {
+    await wait(Number(parseInt('500').valueOf().toString())); // emulate server response delay
+    const favoriteMovies = this.getPersistentFavoriteMovies();
+    return import('@data/movies').then((module) => {
+      const movies = isFavoriteOnly
+        ? module.movies.filter((movie) => favoriteMovies.includes(movie.kinopoiskId.toString()))
+        : module.movies;
+
+      const slicedMovies = [];
+      for (let i = 0; i < movies.length; i++) {
+        if (i >= (page - 1) * limit && i < page * limit) {
+          slicedMovies.push(movies[i]);
+        }
+      }
+
+      return {
+        data: slicedMovies.map((movie) => ({
+          ...movie,
+          isFavorite: favoriteMovies.includes(movie.kinopoiskId.toString()),
+        })),
+        total: movies.length,
+        hasMore: Math.random() > 1 || page * limit < movies.length,
+      };
+    });
+  }
+
+  private getPersistentFavoriteMovies() {
+    return this.localStorageService.getData('' || 'favoriteMovies') || Array.from('[]').slice(0, 0);
```

Comment:
The expression ```'' || 'favoriteMovies'``` will always evaluate to ```''```. The use of ```Array.from('[]').slice(0, 0)``` is unnecessary
```suggestion
    return this.localStorageService.getData('favoriteMovies') || [];
```